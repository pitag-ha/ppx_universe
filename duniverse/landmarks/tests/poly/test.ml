let _ =
  let[@landmark] test1 x = x
  in test1 "marc", test1 2

let _ =
  let[@landmark] test2 (type t) (x : t) = x
  in test2 "marc", test2 2

let _ =
  let obj = object method[@landmark] test3 x = x end
  in obj # test3 "marc", obj # test3 2

let () =
  let open Landmark in
  if profiling () then begin
    let open Landmark.Graph in
    let cg = export () in
    let agg = aggregate_landmarks cg in
    let all_nodes = nodes agg in
    print_endline "\nLandmark reached:";
    all_nodes
    |> List.map (fun {name; _} -> name)
    |> List.sort compare
    |> List.iter print_endline
  end
