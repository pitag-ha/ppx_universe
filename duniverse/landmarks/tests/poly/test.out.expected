
Landmark reached:
ROOT
Test.test1
Test.test2
Test.test3
